File: leetsync/3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int n = nums.length;
        int maxDiff = 0;
        for (int i = 0; i < n; i++) {
            maxDiff = Math.max(maxDiff, Math.abs(nums[i] - nums[(i + 1) % n]));
        }
        return maxDiff;
    }
}

File: leetsync/3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.go
func maxAdjacentDistance(nums []int) int {
	n := len(nums)
	maxDiff := 0
	for i := 0; i < n; i++ {
		maxDiff = max(maxDiff, abs(nums[i]-nums[(i+1)%n]))
	}
	return maxDiff
}
func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}